contrib/drivers/lvm/lvm_metrics.go

```diff
@@ -0,0 +1,217 @@
+// 	metricList-> Posted metric list
+//	supportedMetrics -> list of supported metrics
+func (d *MetricDriver) ValidateMetricsSupportList(metricList []string, resourceType string) (supportedMetrics []string, err error) {
+	var configs Configs
+	//Read supported metric list from yaml config
+	//source, err := ioutil.ReadFile(ConfFile)
+	//Todo: Move this to read from file
+	source := []byte(data)
+	/*if err != nil {
+		log.Errorf("Not able to read file  %s%v", ConfFile, err)
+	}*/
+
+	error := yaml.Unmarshal(source, &configs)
+	if error != nil {
+		log.Fatalf("Unmarshal error: %v", error)
+	}
+
+	for _, resources := range configs.Cfgs {
+		switch resources.Resource {
+		//ToDo: Other Cases needs to be added
+		case "volume":
+			for _, metricName := range metricList {
+				if metricInMetrics(metricName, resources.Metrics) {
+					supportedMetrics = append(supportedMetrics, metricName)
+
+				} else {
+					log.Infof("metric:%s is not in the supported list", metricName)
+				}
+			}
+		}
+	}
+	return supportedMetrics, nil
+}
+
+//	CollectMetrics: Driver entry point to collect metrics. This will be invoked by the dock
+//	metricsList-> posted metric list
+//	instanceID -> posted instanceID
+//	metricArray	-> the array of metrics to be returned
+func (lvm_driver *MetricDriver) CollectMetrics(metricsList []string, instanceID string) (metricArray []model.MetricSpec, err error) {
+
+	// get MEtrics to unit map
+	metric_to_unit_map := getMetricToUnitMap()
+	//validate metric support list
+	supportedMetrics, err := lvm_driver.ValidateMetricsSupportList(metricsList, "volume")
+	if supportedMetrics == nil {
+		log.Infof("No metrics found in the  supported metric list")
+	}
+	metricMap, err := lvm_driver.cli.CollectMetrics(supportedMetrics, instanceID)
+
+	var tempMetricArray []model.MetricSpec
+	for _, element := range metricsList {
+		val, _ := strconv.ParseFloat(metricMap[element], 64)
+		//Todo: See if association  is required here, resource discovery could fill this information
+		associatorMap := make(map[string]string)
+
+		metricValue := model.Metric{
+			Timestamp: getCurrentUnixTimestamp(),
+			Value:     val,
+		}
+		metricValues := make([]model.Metric, 0)
+		metricValues = append(metricValues, metricValue)
+
+		metric := model.MetricSpec{
+			InstanceID: instanceID,
+
+			InstanceName: metricMap["InstanceName"],
+
+			Job: "OpenSDS",
+
+			Associator: associatorMap,
+
+			Source: "Node",
+			//Todo Take Componet from Post call, as of now it is only for volume
+			Component: "Volume",
+
+			Name: element,
+			//Todo : Fill units according to metric type
+			Unit: metric_to_unit_map[element],
+			//Todo : Get this information dynamically ( hard coded now , as all are direct values
+			IsAggregated: false,
+
+			MetricValues: metricValues,
+		}
+		tempMetricArray = append(tempMetricArray, metric)
+	}
+	metricArray = tempMetricArray
+	return
```

Comment:
It's not allowed to return none here, since you defined some variables to be returned.

Reply:
return variables were declared in return type, anyways changes to the slandered return format.